### ISSUE TITLE
Using `exclude:` in `.lintr` results in error

### ISSUE THREAD

**Issue by jabenninghoff** [r-lib/lintr#1401]
When using the 3.0.0 CRAN release of lintr, listing with the following `.lintr` file results in an unexpected error:

```r
linters: linters_with_defaults(
    line_length_linter(100) 
  )
exclude: list(
    "bad.R" = list(
      1, # global exclusions are unnamed
      line_length_linter = 4:6
    )
  )
```

(adapted from the [Get Started](https://lintr.r-lib.org/articles/lintr.html) page)

Error message when running `lint_dir()`:

```
$ Rscript -e 'lintr::lint_dir(".")'
Error in `[.data.frame`(rex::re_matches(lines, exclude, locations = TRUE),  : 
  undefined columns selected
Calls: <Anonymous> ... do.call -> <Anonymous> -> lapply -> FUN -> [ -> [.data.frame
Execution halted
```

To reproduce: run `Rscript -e 'lintr::lint_dir(".")'` from a directory containing the `.lintr` file above, and the following `bad.R`:

```r
tmp = "value"

# regular comment
# really long comment ............................................................................................................
# really long comment ............................................................................................................
# really long comment ............................................................................................................
# regular comment
```

**Comment by AshesITR** [comment]
Hi @jabenninghoff. You're looking for `exclusions: list(...)`.
`exclude: ...` modifies the regex used for the in-file exclusion syntax.

**Comment by AshesITR** [comment]
Sorry for the premature closure - thanks for reporting this bug in the documentation!
